File: 90-subsets-ii/subsets-ii.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        Helper(0,nums,ans,temp);
        return ans;

    }
    public void Helper(int index,int[] nums,List<List<Integer>> ans,List<Integer> temp)
    {
        ans.add(new ArrayList<>(temp));
        if(index== nums.length)
        {
            return;
        }
        for(int i = index ; i< nums.length ; i++)
        {
            if(i!=index && nums[i]==nums[i-1]) continue;
            temp.add(nums[i]);
            Helper(i+1,nums,ans,temp);
            temp.remove(temp.size()-1);
        }
        
    }
}

File: 90-subsets-ii/subsets-ii.cpp
class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums){
       sort(nums.begin(),nums.end());
       vector<vector<int>> ans;
       vector<int> temp;
       Helper(0,nums,temp,ans);
       return ans;
    }
    void Helper(int index, vector<int>& nums,vector<int>& temp, vector<vector<int>>& ans)
    {
        ans.push_back(temp);
        if(index == nums.size())
        { 
            return;
        }
        for( int i = index ; i < nums.size();i++)
        {
             if(i!=index && nums[i]==nums[i-1]) continue;
             temp.push_back(nums[i]);
             Helper(i+1,nums,temp,ans);
             temp.pop_back();
        }
    }
};

